commands/tests/test_upgrade_paths.py

```diff
@@ -0,0 +1,54 @@
+import mock
+import pytest
+
+from leapp.cli.commands.upgrade import upgrade_path
+from leapp.exceptions import CommandError
+
+
+@mock.patch('leapp.cli.commands.upgrade.upgrade_path.get_upgrade_paths_config',
+            return_value={'default': {'7.9': ['8.4'], '8.6': ['9.0'], '7': ['8.4'], '8': ['9.0']}})
+def test_get_target_version(mock_open, monkeypatch):
+
+    monkeypatch.setattr(upgrade_path, 'get_os_release_version_id', lambda x: '8.6')
+    assert upgrade_path.get_target_version('default') == '9.0'
+
+    monkeypatch.setenv('LEAPP_DEVEL_TARGET_RELEASE', '')
+    monkeypatch.setattr(upgrade_path, 'get_os_release_version_id', lambda x: '8.6')
+    assert upgrade_path.get_target_version('default') == '9.0'
+
+    monkeypatch.setenv('LEAPP_DEVEL_TARGET_RELEASE', '1.2.3')
+    assert upgrade_path.get_target_version('default') == '9.0'
+
+    monkeypatch.delenv('LEAPP_DEVEL_TARGET_RELEASE', raising=True)
+    # unsupported path
+    monkeypatch.setattr(upgrade_path, 'get_os_release_version_id', lambda x: '8.5')
+    assert upgrade_path.get_target_version('default') == '9.0'
+
+
+@mock.patch('leapp.cli.commands.upgrade.upgrade_path.get_upgrade_paths_config',
+            return_value={'default': {'7.9': ['8.4'], '8.6': ['9.0'], '7': ['8.4'], '8': ['9.0']}})
+def test_vet_upgrade_path(mock_open, monkeypatch):
+    monkeypatch.setattr(upgrade_path, 'get_os_release_version_id', lambda x: '8.6')
+
+    # make sure env var LEAPP_DEVEL_TARGET_RELEASE takes precedence
+    # when env var set to a bad version - abort the upgrade
+    args = mock.Mock(target='9.0')
+    monkeypatch.setenv('LEAPP_DEVEL_TARGET_RELEASE', '1.2.3shouldfail')
+    with pytest.raises(CommandError):
+        upgrade_path.vet_upgrade_path(args)
```

Comment:
this part of the test is correct. in case of invalid data format, the error should be still raised. But with explanation that "X.Y" format is expected.

Reply:
Should be fixed